src/server/pps/server/api_server.go

```diff
@@ -2125,22 +2125,25 @@ func getExpectedNumWorkers(pipelineInfo *pps.PipelineInfo) (int, error) {
 func (a *apiServer) RerunPipeline(ctx context.Context, request *pps.RerunPipelineRequest) (response *emptypb.Empty, err error) {
 	metricsFn := metrics.ReportUserAction(ctx, a.reporter, "RerunPipeline")
 	defer func(start time.Time) { metricsFn(start, err) }(time.Now())
+	if err != nil {
```

Comment:
should thisbe inside the defer func?

Reply:
It ought to be deleted entirely☺ Good catch.